src/main/java/graphql/execution/MergedField.java

```diff
@@ -171,25 +322,64 @@ public MergedField transform(Consumer<Builder> builderConsumer) {
         return builder.build();
     }
 
+    /**
+     * Runs a consumer for each field
+     *
+     * @param fieldConsumer the consumer to run
+     */
+    public void forEach(Consumer<Field> fieldConsumer) {
+        fieldConsumer.accept(singleField);
+    }
+
     public static class Builder {
 
-        private final ImmutableList.Builder<Field> fields = new ImmutableList.Builder<>();
+        /*
+            The builder logic is complicated by these dual singleton / list duality code,
+            but it prevents memory allocation and every bit counts
+            when the CPU is running hot and an operation has lots of fields!
+         */
+        private ImmutableList.@Nullable Builder<Field> fields;
+        private @Nullable Field singleField;
         private final ImmutableList.Builder<DeferredExecution> deferredExecutions = new ImmutableList.Builder<>();
```

Comment:
The builder has been made more complicated and smarter in memory allocation BUT its not strictly needed.  Because in the graphql-java engine we never call the builder in the field collector any more.  We use helpers that directly create `MergedField`s without a builder.  So the builder code is improved, its not strictly needed